examples/example_emscripten_wgpu/main.cpp

```diff
@@ -11,52 +11,74 @@
 #include "imgui_impl_glfw.h"
 #include "imgui_impl_wgpu.h"
 #include <stdio.h>
+#ifdef __EMSCRIPTEN__
 #include <emscripten.h>
 #include <emscripten/html5.h>
 #include <emscripten/html5_webgpu.h>
+#else
+#include <webgpu/webgpu_glfw.h>
+#endif
 #include <GLFW/glfw3.h>
 #include <webgpu/webgpu.h>
 #include <webgpu/webgpu_cpp.h>
 
 // Global WebGPU required states
-static WGPUDevice        wgpu_device = nullptr;
-static WGPUSurface       wgpu_surface = nullptr;
-static WGPUTextureFormat wgpu_preferred_fmt = WGPUTextureFormat_RGBA8Unorm;
-static WGPUSwapChain     wgpu_swap_chain = nullptr;
-static int               wgpu_swap_chain_width = 0;
-static int               wgpu_swap_chain_height = 0;
+static wgpu::Instance      wgpu_instance = nullptr;
+static wgpu::Surface       wgpu_surface = nullptr;
+static wgpu::Device        wgpu_device = nullptr;
```

Comment:
As others have said, there's no need to use C++ wrappers here since they don't simplify much in this example.